File: app/src/main/java/com/example/fluke/mvp_project/main/Presenter.java
package com.example.fluke.mvp_project.main;


import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.widget.Toast;
import java.util.ArrayList;

/**
 * Created by Fluke on 11/1/2561.
 */

public class Presenter implements InterMainActivity.Presenter {

    private InterMainActivity.View view;
    private  Context context;

    public Presenter(InterMainActivity.View view , Context context) {
        this.view = view;
    }

    @Override
    public void viewOncreate() {

    }

    @Override
    public void viewOnStart() {

    }

    @Override
    public void viewOnResume() {

    }

    @Override
    public void viewOnPause() {

    }

    @Override
    public void viewOnStop() {

    }

    @Override
    public void viewOnDestroy() {

    }

    @Override
    public void viewOnCreateView() {

    }

    @Override
    public void onButtonClick() {
        String name ="ไม่ได้ระบุชื่อ";
        String choco="";
        String qunn="ไม่ได้ระบุจำนวน";
        String cream="";

        if (view.getName() != null){
            name = view.getName();
        }

        if (view.getQunn()+"" != null){
            qunn   = view.getQunn()+"";
        }

        if (view.getChoco() != null){
           choco = view.getChoco();
        }
        if (view.getCream() != null){
            cream = view.getCream();
        }

        ArrayList<String> itemList= new ArrayList<>();
        if (!name.equals("")){
            itemList.add(name);
        }
        if (!choco.equals("")){
            itemList.add(choco);
        }
        if (!cream.equals("")){
            itemList.add(cream);
        }
        if (!qunn.equals("")){
            itemList.add(qunn);
        }

        String resultList="";
        for (int i = 0 ; i< itemList.size() ; i++){
            if (i==0){
                resultList+=itemList.get(i)+" สั่งกาแฟ\n\n";
            }else
            if (i==itemList.size()-1){
                resultList+="จำนวน "+itemList.get(i)+" แก้ว";
            }else{
                resultList+="เพิ่ม "+itemList.get(i)+"\n\n";
            }
        }
{
            Intent i = new Intent(Intent.ACTION_SEND);
            i.setType("message/rfc822");
            i.putExtra(Intent.EXTRA_EMAIL  , new String[]{"example@example.com"});
            i.putExtra(Intent.EXTRA_SUBJECT, "D JAAAA EIEI");
            i.putExtra(Intent.EXTRA_TEXT   ,"คุณ "+resultList);
            try {
                view.getContextx().startActivity(Intent.createChooser(i, "Send mail..."));
            } catch (android.content.ActivityNotFoundException ex) {
                Toast.makeText(context, "There are no email clients installed.", Toast.LENGTH_SHORT).show();
            }
            Log.e("apwefawef",view.getName());
        }

    }
}

File: app/src/main/java/com/example/fluke/mvp_project2_kotlin/tasking/Presenter.kt
package com.example.fluke.mvp_project2_kotlin.tasking

import android.content.Context
import android.util.Log
import android.view.Display
import android.widget.Toast


/**
 * Created by Fluke on 12/1/2561.
 */

class Presenter : Contacter.Present {
    override fun OnClicked(model: Models, context: Context, view: Contacter.View) {
        if (model.name.isNullOrEmpty() || model.name.equals(" ") || model.Quantity== "0"
                || model.name.equals("")) {

            Toast.makeText(context, "กรุณากรอกข้อมูลให้ครบด้วยครับ", Toast.LENGTH_SHORT).show()

        } else {

            view.SendIntents(OnDataReady(model.name.toString(), model.isCream.toString(),
                    model.isChoco.toString(), model.Quantity.toString()))
            Log.e("finalText", OnDataReady(model.name.toString(), model.isCream.toString()
                    , model.isChoco.toString(), model.Quantity.toString()))

        }
    }


     fun OnDataReady(name: String , isIce: String , isChoco: String, qunn: String): String {
        var resuld: String = "คุณ $name \nได้ทำการสั่งกาแฟ"
        if (isIce == "true") {
            resuld += "\nเพิ่ม Whipping cream"
        }
        if (isChoco == "true") {
            resuld += "\nเพิ่ม Chocolate"
        }
        resuld += "\n จำนวน $qunn แก้ว"
        return resuld

    }

    override fun CheckButNegative(quans: Int): Int {
        var quan: Int = quans - 1
        return if (quans < 1) {
            0
        } else {
            quan
        }
    }
}